test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/concurrent/ConcurrentHashMapTest.scala

```diff
@@ -50,6 +50,13 @@ class ConcurrentHashMapTest extends MapTest {
     assertEquals("one", chm.get("ONE"))
   }
 
+  @Test def putIfAbcentShould_issue_2539(): Unit = {
+    val chm = factory.empty[String, String]
+    assertNull(chm.putIfAbsent("abc", "def"))
+    assertEquals("def", chm.putIfAbsent("abc", "def"))
+    assertEquals("def", chm.putIfAbsent("abc", "ghi"))
+    assertEquals("def", chm.putIfAbsent("abc", "jkl"))
```

Comment:
Maybe worth testing `putIfAbsent` with something else than `"abc"` and verify that it returns `null` again, for a different key.